### PR TITLE
Move more functions from common pkg to dedicated packages

### PR DESCRIPTION
**Pull request by rfranzke**
**How to categorize this PR?**

/area dev-productivity
/kind cleanup

**What this PR does / why we need it**:
With this PR we are moving more functions from the `common` package to better places, i.e., dedicated packages that provide a better 🏡 for them :)

/squash

**Release note**:

```other operator
NONE
```

### CODE REVIEW

**Comment by timebertt** [comment]
```
Executing gofmt/goimports
Unformatted files detected:
pkg/utils/kubernetes/statefulset.go
make: *** [Makefile:169: check] Error 1
```

**Comment by rfranzke** [comment]
@timebertt fixed

File: pkg/extensions/cluster.go [hunk 1]
```diff
@@ -27,13 +27,84 @@ import (
...
+		seedObj         *gardencorev1beta1.Seed
+		shootObj        *gardencorev1beta1.Shoot
+	)
+
+	if cloudProfile != nil {
+		cloudProfileObj = cloudProfile.DeepCopy()
+		cloudProfileObj.TypeMeta = metav1.TypeMeta{
+			APIVersion: gardencorev1beta1.SchemeGroupVersion.String(),
+			Kind:       "CloudProfile",
+		}
```
Comment (vpnachev) [inline pkg/extensions/cluster.go:76]: What do you think about purging the `ManagedFields` here? (also for the seed and shoot).
```suggestion
		cloudProfileObj = cloudProfile.DeepCopy()
		cloudProfileObj.TypeMeta = metav1.TypeMeta{
			APIVersion: gardencorev1beta1.SchemeGroupVersion.String(),
			Kind:       "CloudProfile",
		}
		cloudProfileObj.ObjectMeta.ManagedFields = nil
```

File: pkg/utils/kubernetes/deployment.go [hunk 2]
```diff
@@ -15,11 +15,13 @@
...
 import (
+	"context"
 	"fmt"
 	"strings"
 
 	apierrors "k8s.io/apimachinery/pkg/api/errors"
 	"k8s.io/apimachinery/pkg/labels"
+	"sigs.k8s.io/controller-runtime/pkg/client"
 
 	"github.com/Masterminds/semver"
```
Comment (vpnachev) [inline pkg/utils/kubernetes/deployment.go]:
```suggestion
	"github.com/Masterminds/semver"
	apierrors "k8s.io/apimachinery/pkg/api/errors"
	"k8s.io/apimachinery/pkg/labels"
	"sigs.k8s.io/controller-runtime/pkg/client"
```

File: pkg/utils/gardener/dns.go [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+	// a Shoot cluster. For example, when a Shoot specifies domain 'cluster.example.com', the ingress domain would be
+	// '*.<IngressPrefix>.cluster.example.com'.
+	IngressPrefix = "ingress"
+	// InternalDomainKey is a key which must be present in an internal domain constructed for a Shoot cluster. If the
+	// configured internal domain already contains it, it won't be added twice. If it does not contain it, it will be
+	// appended.
+	InternalDomainKey = "internal"
+)
+
+// GetDomainInfoFromAnnotations returns the provider and the domain that is specified in the give annotations.
```
Comment (vpnachev) [inline pkg/utils/gardener/dns.go]:
```suggestion
// GetDomainInfoFromAnnotations returns the provider and the domain that is specified in the given annotations.
```

File: pkg/operation/botanist/component/extensions/backupentry/backupentry.go [hunk 4]
```diff
@@ -18,18 +18,18 @@ import (
...
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	"k8s.io/apimachinery/pkg/runtime"
 	"sigs.k8s.io/controller-runtime/pkg/client"
 	"sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
+
+	gardencorev1alpha1 "github.com/gardener/gardener/pkg/apis/core/v1alpha1"
+	v1beta1constants "github.com/gardener/gardener/pkg/apis/core/v1beta1/constants"
+	extensionsv1alpha1 "github.com/gardener/gardener/pkg/apis/extensions/v1alpha1"
+	"github.com/gardener/gardener/pkg/extensions"
+	"github.com/gardener/gardener/pkg/operation/botanist/component"
```
Comment (vpnachev) [inline pkg/operation/botanist/component/extensions/backupentry/backupentry.go]: Why the import from the same repo are now the third block?
I know, we are not consistent across all files, but I am wondering why it is changed now.

File: pkg/operation/botanist/component/extensions/network/network.go [hunk 5]
```diff
@@ -19,17 +19,17 @@ import (
...
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	"k8s.io/apimachinery/pkg/runtime"
 	"sigs.k8s.io/controller-runtime/pkg/client"
 	"sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
+
+	"github.com/gardener/gardener/pkg/apis/core/v1alpha1"
+	v1beta1constants "github.com/gardener/gardener/pkg/apis/core/v1beta1/constants"
+	extensionsv1alpha1 "github.com/gardener/gardener/pkg/apis/extensions/v1alpha1"
+	"github.com/gardener/gardener/pkg/extensions"
+	"github.com/gardener/gardener/pkg/operation/botanist/component"
```
Comment (vpnachev) [inline pkg/operation/botanist/component/extensions/network/network.go]: Same as above.

**Review by vpnachev — COMMENTED** [review]
Thanks!

Overall looks good to me. I just have some questions and minor suggestions.

File: pkg/operation/botanist/component/extensions/backupentry/backupentry.go (hunk 4 above)
Comment (rfranzke) [reply pkg/operation/botanist/component/extensions/backupentry/backupentry.go]: This happens sometimes when you're working with GoLand IDE, I guess it's the default setting (I've also seen it in PRs from other authors). Anyways, I've reverted it for now as I was anyways on it (addressing your other suggestions).

Comment (vpnachev) [reply pkg/operation/botanist/component/extensions/backupentry/backupentry.go]: Thanks! I will have in mind.

**Review by vpnachev — APPROVED** [review]
/lgtm

**Pull request merged**